libgcc/config/riscv/save-restore.S

```diff
@@ -294,6 +294,40 @@ __riscv_restore_0:
 
 #else
 
+#ifdef __riscv_32e
+__riscv_save_2:
+__riscv_save_1:
+__riscv_save_0:
+  .cfi_startproc
+  # __riscv_save_* routine use t0/x5 as return address
+  .cfi_return_column 5
+  addi sp, sp, -16
+  .cfi_def_cfa_offset 16
+  sw s1, 4(sp)
+  .cfi_offset 9, -16
+  sw s0, 8(sp)
+  .cfi_offset 8, -8
+  sw ra, 12(sp)
+  .cfi_offset 1, -4
+  jr t0
+  .cfi_endproc
+
+__riscv_restore_2:
+__riscv_restore_1:
+__riscv_restore_0:
+  .cfi_startproc
+  .cfi_def_cfa_offset 16
+  lw s1, 4(sp)
+  .cfi_restore 9
+  lw s0, 8(sp)
+  .cfi_restore 8
+  lw ra, 12(sp)
+  .cfi_restore 1
+  addi sp, sp, 16
+  .cfi_def_cfa_offset 0
+  ret
+  .cfi_endproc
+#else
```

Comment:
Since RVE only has a 4-byte stack alignment, and we are only saving 3 registers, I think we should only adjust SP by 12 bytes, not 16.